pkg/controller/controller_instance_test.go

```diff
@@ -2370,3 +2385,318 @@ func TestPollInstanceUsingOriginatingIdentity(t *testing.T) {
 		}()
 	}
 }
+
+func TestReconcileServiceInstanceWithHTTPStatusCodeErrorOrphanMitigation(t *testing.T) {
+	cases := []struct {
+		name                     string
+		statusCode               int
+		triggersOrphanMitigation bool
+	}{
+		{
+			name:                     "Status OK",
+			statusCode:               200,
+			triggersOrphanMitigation: false,
+		},
+		{
+			name:                     "other 2XX",
+			statusCode:               201,
+			triggersOrphanMitigation: true,
+		},
+		{
+			name:                     "3XX",
+			statusCode:               300,
+			triggersOrphanMitigation: false,
+		},
+		{
+			name:                     "408",
+			statusCode:               408,
+			triggersOrphanMitigation: true,
+		},
+		{
+			name:                     "other 4XX",
+			statusCode:               400,
+			triggersOrphanMitigation: false,
+		},
+		{
+			name:                     "5XX",
+			statusCode:               500,
+			triggersOrphanMitigation: true,
+		},
+	}
+
+	for _, tc := range cases {
+		_, fakeCatalogClient, _, testController, sharedInformers := newTestController(t, fakeosb.FakeClientConfiguration{
+			ProvisionReaction: &fakeosb.ProvisionReaction{
+				Error: osb.HTTPStatusCodeError{
+					StatusCode: tc.statusCode,
+				},
+			},
+		})
+
+		sharedInformers.ServiceBrokers().Informer().GetStore().Add(getTestServiceBroker())
+		sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+		sharedInformers.ServicePlans().Informer().GetStore().Add(getTestServicePlan())
+
+		instance := getTestServiceInstance()
+
+		err := testController.reconcileServiceInstance(instance)
+
+		// The action should be:
+		// 0. Updating the status
+		actions := fakeCatalogClient.Actions()
+		if ok := expectNumberOfActions(t, tc.name, actions, 2); !ok {
+			continue
```

Comment:
These tests can be a little cleaner if you use sub-tests via `t.Run`. That way we don't have to sprinkle `continue`s everywhere. We also wouldn't need to pass a fail func into the testOrphanMitigationInProgress function.

Reply:
Would you be okay with that in a follow-up? I was following the precedent set by functions like `TestUpdateServiceInstanceCondition` and `TestUpdateServiceBrokerCondition`

Reply:
Yep. I am fine with doing nothing. It was just a suggestion.